### PR TITLE
[typescript-graphql-request] Allow passing Abortsignal to graphql-request

### PR DESCRIPTION
**Pull request by shkreios**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Related #31 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- repository tests

## Checklist:

- [x] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shkreios** [comment]
I moved this [pr](https://github.com/dotansimha/graphql-code-generator/pull/8361) from the main repo, but have not yet tested the version of this code in any private repo. If you have some time @yemi you can test the code from this pr. Thank you.

File: packages/plugins/typescript/graphql-request/src/visitor.ts [hunk 1]
```diff
@@ -116,20 +116,32 @@ export class GraphQLRequestVisitor extends ClientSideBaseVisitor<
           }
           return `${operationName}(variables${optionalVariables ? '?' : ''}: ${
             o.operationVariablesTypes
-          }, requestHeaders?: Dom.RequestInit["headers"]): Promise<{ data: ${o.operationResultType}; extensions?: ${
-            this.config.extensionsType
-          }; headers: Dom.Headers; status: number; }> {
-    return withWrapper((wrappedRequestHeaders) => client.rawRequest<${
-      o.operationResultType
-    }>(${docArg}, variables, {...requestHeaders, ...wrappedRequestHeaders}), '${operationName}', '${operationType}');
+          }, signal?: AbortSignal, requestHeaders?: Dom.RequestInit["headers"]): Promise<{ data: ${
```
Comment (ardatan) [inline packages/plugins/typescript/graphql-request/src/visitor.ts:119]: What about accepting `RequestInit` in general instead of adding each parameter one by one

**Comment by shkreios** [comment]
hey @ardatan @vadimdemedes sorry it took me longer than expected to get back to this.
I wanted to avoid pushing anything that i did not at least test once inside an example repo. While following the [getting started](https://the-guild.dev/graphql/codegen/docs/getting-started) of new client preset of GraphQL codegen, I noticed that generating any library-specific code is not advised anymore and instead only `TypedDocumentNode` should be used. 

**TLDR:**
The codegen approach of this plugin is deprecated. Therefore, I don't know if it even makes to update it. 
Here is an [example repo](https://github.com/shkreios/graphql-codegen-tanstack-query-graphql-request-example) & [stackblitz](https://stackblitz.com/github/shkreios/graphql-codegen-tanstack-query-graphql-request-example) where I tested passing the `Abortsignal` to `graphql-request` from `tanstack/query` with the new graphql-codegen/client-preset

Will keep this pr open to wait for feedback

**Comment by vadimdemedes** [comment]
> The codegen approach of this plugin is deprecated

How come? What's a new approach?

**Comment by shkreios** [comment]
Hey, I took another look at the [options](https://github.com/prisma-labs/graphql-request/blob/master/src/types.ts#L92) that `graphql-options` allows. As you can see, `headers` & `signal` are the only options allowed. If we want pass any other options from `RequestInit` to it we need to open a pr for `graphql-request`.

**Comment by shkreios** [comment]
Test repo with this pr [repo](https://github.com/shkreios/graphql-codegen-tanstack-query-graphql-request-example/tree/with-graphql-request-plugin) & [stackblitz](https://stackblitz.com/github/shkreios/graphql-codegen-tanstack-query-graphql-request-example/tree/with-graphql-request-plugin)

**Comment by mp205** [comment]
any update on this?

**Comment by shkreios** [comment]
> any update on this?

@mp205 From my side, the pr is ready for review/merge. As mentioned in my [comment here](https://github.com/dotansimha/graphql-code-generator-community/pull/32#issuecomment-1365507625), I don't know how relevant it still is as `graphql-codegen` does advise against generating lib-specific code and instead to use `TypedDocumentNodes`. 

And as mentioned in [this comment ](https://github.com/dotansimha/graphql-code-generator-community/pull/32#issuecomment-1366578192), the proposed changes by @ardatan & @vadimdemedes are not possible as `graphql-request `only allows passing `signal` and `headers` to it instead of the `Dom.RequestInit` as a whole.

**Comment by happyeric77** [comment]
Hello, Just would like to know will the PR be merged anytime soon.

**Comment by mxswat** [comment]
Hi, is there any updates regarding this MR? This feature would be crucial to my current project.

**Comment by tranhoanvo** [comment]
@ardatan, @vadimdemedes Hi, can you help to let this be merged ?

**Comment by happyeric77** [comment]
I am checking in here to see if we plan to merge this. @shkreios @ardatan
